tests/diff-folded.rs

```diff
@@ -0,0 +1,159 @@
+#[macro_use]
+extern crate pretty_assertions;
+
+extern crate inferno;
+
+use inferno::diff_folded::{self, Options};
+use log::Level;
+use std::fs::{self, File};
+use std::io::{self, BufRead, BufReader, Cursor};
+
+fn test_diff_folded(
+    infile1: &str,
+    infile2: &str,
+    expected_result_file: &str,
+    options: Options,
+) -> io::Result<()> {
+    let metadata = match fs::metadata(expected_result_file) {
+        Ok(m) => m,
+        Err(e) => {
+            if e.kind() == io::ErrorKind::NotFound {
+                // be nice to the dev and make the file
+                let mut f = File::create(expected_result_file).unwrap();
+                diff_folded::from_files(&options, &infile1, &infile2, &mut f)?;
+                fs::metadata(expected_result_file).unwrap()
+            } else {
+                return Err(e.into());
+            }
+        }
+    };
+
+    let expected_len = metadata.len() as usize;
+    let mut result = Cursor::new(Vec::with_capacity(expected_len));
+    let return_value = diff_folded::from_files(&options, infile1, infile2, &mut result)?;
+    let expected = BufReader::new(File::open(expected_result_file).unwrap());
+    // write out the expected result to /tmp for easy restoration
+    result.set_position(0);
+    let rand: u64 = rand::random();
+    let tm = std::env::temp_dir().join(format!("test-{}.svg", rand));
+    if fs::write(&tm, result.get_ref()).is_ok() {
+        eprintln!("test output in {}", tm.display());
+    }
+    // and then compare
+    result.set_position(0);
+    compare_results(result, expected, expected_result_file);
+    Ok(return_value)
+}
```

Comment:
We really ought to find a way to share this code across all of our different tests. I _think_ it should be doable by passing in a closure that goes from input(s) to output.

Reply:
Yes, I've thought of doing that. I just haven't gotten to it yet.

Reply:
I'm leaving this for a separate PR. There's enough differences among `flamegraph`, `collapse`, and `diff-folded` that make it a bit complicated.

- Collapse uses `collapser.collapse` if the input file is a gzip, but `collapser.collapse_file` otherwise.
- They all have incompatible implementations of `compare_results`.
- `diff-folded` requires exactly two input files, but the others don't.

It would be nice to share code across these tests, but I don't think we can do it with a single closure. Maybe we could solve it using a trait?